File: src/main/java/de/gieskerb/class_craft/classes/LumberjackClass.java
package de.gieskerb.class_craft.classes;

import de.gieskerb.class_craft.data.HorseData;
import org.bukkit.DyeColor;
import org.bukkit.Material;
import org.bukkit.block.banner.Pattern;
import org.bukkit.block.banner.PatternType;
import org.bukkit.enchantments.Enchantment;
import org.bukkit.entity.Horse;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemFlag;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.BannerMeta;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;
import org.json.simple.JSONObject;

import java.util.ArrayList;
import java.util.List;

public class LumberjackClass extends BaseClass {

    public final static String CLASS_NAME = "Lumberjack";
    private static ItemStack displayItem = null;

    public LumberjackClass(Player player) {
        super(LumberjackClass.CLASS_NAME, player);
    }

    public LumberjackClass(JSONObject json, Player player) {
        super(json, player);
    }

    public static ItemStack getDisplayItem() {
        if (displayItem == null) {
            displayItem = new ItemStack(Material.NETHERITE_AXE);
            var itemMeta = displayItem.getItemMeta();
            assert itemMeta != null;
            itemMeta.setItemName("§6§lLumberjack-Class");
            List<String> loreList = new ArrayList<>();
            loreList.add("§r§2super is a lore line!");
            loreList.add("§r§2And super is anoter one.");
            itemMeta.setLore(loreList);
            itemMeta.addItemFlags(ItemFlag.HIDE_ATTRIBUTES);
            displayItem.setItemMeta(itemMeta);
        }
        return displayItem;
    }

    @Override
    public void reapplyRewardEffects() {
        if (super.getLevel() >= 17) {
            super.playerReference.addPotionEffect(new PotionEffect(PotionEffectType.STRENGTH, -1, 1, false, false));
        } else if (super.getLevel() >= 5) {
            super.playerReference.addPotionEffect(new PotionEffect(PotionEffectType.STRENGTH, -1, 0, false, false));
        }
        if (super.getLevel() >= 15) {
            super.playerReference.addPotionEffect(new PotionEffect(PotionEffectType.SLOWNESS, -1, 0, false, false));
        }
    }

    @Override
    public void removePermanentEffects() {
        super.playerReference.removePotionEffect(PotionEffectType.RESISTANCE);
        super.playerReference.removePotionEffect(PotionEffectType.SLOWNESS);
    }

    @Override
    public ItemStack getClassItem() {
        return getDisplayItem();
    }

    @Override
    protected HorseData getHorseData() {
        return new HorseData(Horse.Color.values()[(int) (Math.random() * Horse.Color.values().length)],
                             Horse.Style.values()[(int) (Math.random() * Horse.Style.values().length)],
                             0.3,
                             0.6,
                             60,
                             HorseData.getHorseName(super.playerReference, super.CLASS_NAME),
                             Material.DIAMOND_HORSE_ARMOR);
    }

    @Override
    protected void giveBannerReward() {
        ItemStack item = new ItemStack(Material.GRAY_BANNER);
        BannerMeta m = (BannerMeta) item.getItemMeta();

        List<Pattern> patterns = new ArrayList<>();
        patterns.add(new Pattern(DyeColor.RED, PatternType.FLOWER));
        patterns.add(new Pattern(DyeColor.RED, PatternType.STRIPE_TOP));
        patterns.add(new Pattern(DyeColor.WHITE, PatternType.STRIPE_BOTTOM));
        patterns.add(new Pattern(DyeColor.BLACK, PatternType.STRIPE_MIDDLE));

        m.setPatterns(patterns);

        item.setItemMeta(m);
        super.playerReference.getInventory().addItem(item);
    }

    @Override
    protected void giveFirstToolReward() {
        ItemStack item = new ItemStack(Material.STONE_SWORD);
        item.addUnsafeEnchantment(Enchantment.SHARPNESS, 1);
        item.addUnsafeEnchantment(Enchantment.UNBREAKING, 1);
        super.playerReference.getInventory().addItem(item);
    }

    @Override
    protected void giveSecondToolReward() {
        ItemStack item = new ItemStack(Material.DIAMOND_SWORD);
        item.addUnsafeEnchantment(Enchantment.SHARPNESS, 3);
        item.addUnsafeEnchantment(Enchantment.UNBREAKING, 2);
        item.addUnsafeEnchantment(Enchantment.LOOTING, 1);
        super.playerReference.getInventory().addItem(item);
    }

    @Override
    protected void giveThirdToolReward() {
        ItemStack item = new ItemStack(Material.NETHERITE_SWORD);
        item.addUnsafeEnchantment(Enchantment.SHARPNESS, 5);
        item.addUnsafeEnchantment(Enchantment.SMITE, 5);
        item.addUnsafeEnchantment(Enchantment.BANE_OF_ARTHROPODS, 5);
        super.playerReference.getInventory().addItem(item);
    }
}

File: src/main/kotlin/de/gieskerb/class_craft/classes/LumberjackClass.kt
package de.gieskerb.class_craft.classes

import de.gieskerb.class_craft.data.HorseData
import org.bukkit.DyeColor
import org.bukkit.Material
import org.bukkit.block.banner.Pattern
import org.bukkit.block.banner.PatternType
import org.bukkit.enchantments.Enchantment
import org.bukkit.entity.Horse
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemFlag
import org.bukkit.inventory.ItemStack
import org.bukkit.inventory.meta.BannerMeta
import org.bukkit.potion.PotionEffect
import org.bukkit.potion.PotionEffectType
import org.json.simple.JSONObject

class LumberjackClass : BaseClass {
    constructor(player: Player) : super(Companion.CLASS_NAME, player)

    constructor(json: JSONObject, player: Player) : super(json, player)

    override fun reapplyRewardEffects() {
        if (super.level >= 17) {
            super.playerReference.addPotionEffect(PotionEffect(PotionEffectType.STRENGTH, -1, 1, false, false))
        } else if (super.level >= 5) {
            super.playerReference.addPotionEffect(PotionEffect(PotionEffectType.STRENGTH, -1, 0, false, false))
        }
        if (super.level >= 15) {
            super.playerReference.addPotionEffect(PotionEffect(PotionEffectType.SLOWNESS, -1, 0, false, false))
        }
    }

    override fun removePermanentEffects() {
        super.playerReference.removePotionEffect(PotionEffectType.RESISTANCE)
        super.playerReference.removePotionEffect(PotionEffectType.SLOWNESS)
    }

    override val classItem: ItemStack?
        get() = displayItem

    override val horseData: HorseData
        get() = HorseData(
            Horse.Color.entries.toTypedArray()[(Math.random() * Horse.Color.entries.toTypedArray().size).toInt()],
            Horse.Style.entries.toTypedArray()[(Math.random() * Horse.Style.entries.toTypedArray().size).toInt()],
            0.3,
            0.6,
            60.0,
            HorseData.horseNameByPlayer(super.playerReference, super.CLASS_NAME),
            Material.DIAMOND_HORSE_ARMOR
        )

    override fun giveBannerReward() {
        val item = ItemStack(Material.WHITE_BANNER)
        val m = item.itemMeta as BannerMeta

        val patterns: MutableList<Pattern> = java.util.ArrayList()
        patterns.add(Pattern(DyeColor.GRAY, PatternType.CROSS))
        patterns.add(Pattern(DyeColor.WHITE, PatternType.CIRCLE))
        patterns.add(Pattern(DyeColor.WHITE, PatternType.RHOMBUS))
        patterns.add(Pattern(DyeColor.GRAY, PatternType.TRIANGLE_BOTTOM))
        patterns.add(Pattern(DyeColor.GRAY, PatternType.TRIANGLE_TOP))
        patterns.add(Pattern(DyeColor.GRAY, PatternType.SQUARE_BOTTOM_LEFT))
        patterns.add(Pattern(DyeColor.BROWN, PatternType.STRAIGHT_CROSS))
        patterns.add(Pattern(DyeColor.WHITE, PatternType.STRIPE_MIDDLE))
        patterns.add(Pattern(DyeColor.GRAY, PatternType.STRIPE_RIGHT))
        patterns.add(Pattern(DyeColor.GRAY, PatternType.STRIPE_TOP))

        m.patterns = patterns

        item.setItemMeta(m)
        super.playerReference.inventory.addItem(item)
    }

    override fun giveFirstToolReward() {
        val item = ItemStack(Material.STONE_SWORD)
        item.addUnsafeEnchantment(Enchantment.SHARPNESS, 1)
        item.addUnsafeEnchantment(Enchantment.UNBREAKING, 1)
        super.playerReference.inventory.addItem(item)
    }

    override fun giveSecondToolReward() {
        val item = ItemStack(Material.DIAMOND_SWORD)
        item.addUnsafeEnchantment(Enchantment.SHARPNESS, 3)
        item.addUnsafeEnchantment(Enchantment.UNBREAKING, 2)
        item.addUnsafeEnchantment(Enchantment.LOOTING, 1)
        super.playerReference.inventory.addItem(item)
    }

    override fun giveThirdToolReward() {
        val item = ItemStack(Material.NETHERITE_SWORD)
        item.addUnsafeEnchantment(Enchantment.SHARPNESS, 5)
        item.addUnsafeEnchantment(Enchantment.SMITE, 5)
        item.addUnsafeEnchantment(Enchantment.BANE_OF_ARTHROPODS, 5)
        super.playerReference.inventory.addItem(item)
    }

    companion object {
        const val CLASS_NAME: String = "Lumberjack"
        var displayItem: ItemStack? = null
            get() {
                if (field == null) {
                    field = ItemStack(Material.NETHERITE_AXE)
                    val itemMeta = checkNotNull(field!!.itemMeta)
                    itemMeta.setItemName("§6§lLumberjack-Class")
                    val loreList: MutableList<String> = ArrayList()
                    loreList.add("§r§2super is a lore line!")
                    loreList.add("§r§2And super is anoter one.")
                    itemMeta.lore = loreList
                    itemMeta.addItemFlags(ItemFlag.HIDE_ATTRIBUTES)
                    field!!.setItemMeta(itemMeta)
                }
                return field
            }
            private set
    }
}